### PR TITLE
MAT-6487 admin api to adjust measure version values

### PR DESCRIPTION
**Pull request by sb-prateekkeerthi**
## MADiE PR

Jira Ticket: [MAT-6487](https://jira.cms.gov/browse/MAT-6487)
(Optional) Related Tickets:

### Summary
admin api to adjust measure version values

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included.
* [ ] No extemporaneous files are included (i.e Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

**Review by mcmcphillips — APPROVED** [review]

File: src/main/java/cms/gov/madie/measure/resources/AdminController.java [hunk 1]
```diff
@@ -171,6 +178,128 @@ public ResponseEntity<List<Map<String, Object>>> getMeasureSharedWith(
...
+      @Value("${admin-api-key}") String apiKey,
+      Principal principal,
+      @PathVariable String id,
+      @RequestParam String inCorrectVersion,
+      @RequestParam String correctVersion,
+      @RequestParam String draftVersion) {
+
+    // can version number be anything : eg: 430.0.001 or does it have to follow rules (like it
+    // should be draft of measureSetId)
+    // clear: for qdm: cqmMeasure, export, for qiCore: export
```
Comment (adongare) [inline src/main/java/cms/gov/madie/measure/resources/AdminController.java:194]: were these the questions for POs?

File: src/main/java/cms/gov/madie/measure/resources/AdminController.java [hunk 2]
```diff
@@ -171,6 +178,128 @@ public ResponseEntity<List<Map<String, Object>>> getMeasureSharedWith(
...
+    List<Measure> relatedMeasures =
+        measureRepository.findAllByMeasureSetIdInAndActiveAndMeasureMetaDataDraft(
+            List.of(measureToCorrectVersion.getMeasureSetId()), true, true);
+
+    if (!CollectionUtils.isEmpty(relatedMeasures)
+        && !relatedMeasures.get(0).getId().equals(measureToCorrectVersion.getId())) {
+      throw new MeasureNotDraftableException(measureToCorrectVersion.getId());
+    }
+
+    // check if the draftVersion is less that correctVersion
```
Comment (adongare) [inline src/main/java/cms/gov/madie/measure/resources/AdminController.java:212]:
```suggestion
    // check if the draftVersion is less than correctVersion
```

File: src/main/java/cms/gov/madie/measure/resources/AdminController.java (hunk 1 above)
Comment (sb-prateekkeerthi) [reply src/main/java/cms/gov/madie/measure/resources/AdminController.java:194]: Oh my bad these were the question for POs, let me remove them

File: src/main/java/cms/gov/madie/measure/resources/AdminController.java (hunk 2 above)
Comment (sb-prateekkeerthi) [reply src/main/java/cms/gov/madie/measure/resources/AdminController.java:212]: thank you I'll change that

**Pull request merged**